### PR TITLE
Adding a model stealing demo

### PR DESCRIPTION
**Pull request by TS-Lee**
# Description

This PR adds a model stealing demo which includes use of knockoff net and CopyCat CNN. The notebook also includes the use of reverse sigmoid defense. The reverse sigmoid defense assumes the attacker's using probability values, so I added an option to use probability values in these attacks.

## Type of change

Please check all relevant options.

- [x] Improvement (non-breaking)
- [ ] Bug fix (non-breaking)
- [x] New feature (non-breaking)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Testing

Please describe the tests that you ran to verify your changes. Consider listing any relevant details of your test configuration.

- [ ] Test A
- [ ] Test B

**Test Configuration**:
- OS: Ubuntu 18.04.5 LTS (GNU/Linux 4.15.0-128-generic x86_64)
- Python version: 3.7.9
- ART version or commit number: e28c3eda9679951331a06f199a1a928af107959c
- TensorFlow / Keras / PyTorch / MXNet version: 2.4.0 / 2.4.3 / 1.6.0+cu92 / NA

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by beat-buesser** [comment]
Hi @TS-Lee This is a great idea! It also solves a recent feature request from a user of ART in #756

**Comment by beat-buesser** [comment]
Btw, we also have a README for notebook where you can list the new notebook: https://github.com/Trusted-AI/adversarial-robustness-toolbox/blob/main/notebooks/README.md

**Comment by TS-Lee** [comment]
I'm creating a new PR for this for some cleaning up..
